### DISCUSSION TITLE
How to integrate specmatic with java 11 and spring boot 2.2.6

### DISCUSSION THREAD

**Discussion by prateekkarmakar** [specmatic/specmatic#1493] [Ask For Help] [answered]
I am trying to integrate specmatic with java 11 project and spring boot 2.2.6. I am getting the below error. Which version of specmatic need to be included so that the following error can be mitigated. I am currently using version 2.0.49 of specmatic. Also including the full log and project for details.
[build_out_java11_junit4.txt](https://github.com/user-attachments/files/18228322/build_out_java11_junit4.txt)
[TestProject_11.zip](https://github.com/user-attachments/files/18228336/TestProject_11.zip)

[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 14.529 s
[INFO] Finished at: 2024-12-23T15:28:25+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile (default-testCompile) on project demo: Compilation failure: Compilation failure: 
[ERROR] /D:/SPECMATIC_ERROR_DEMO/demo(2)/demo/src/test/java/com/example/demo/ContractTest.java:[10,25] cannot access io.specmatic.test.SpecmaticJUnitSupport
[ERROR]   bad class file: C:\Users\KARMAKAR(S)\.m2\repository\io\specmatic\junit5-support\2.0.49\junit5-support-2.0.49.jar(/io/specmatic/test/SpecmaticJUnitSupport.class)
[ERROR]     class file has wrong version 61.0, should be 55.0
[ERROR]     Please remove or make sure it appears in the correct subdirectory of the classpath.
[ERROR] /D:/SPECMATIC_ERROR_DEMO/demo(2)/demo/src/test/java/com/example/demo/ContractTest.java:[16,35] cannot find symbol
[ERROR]   symbol: class SpecmaticJUnitSupport
[ERROR] /D:/SPECMATIC_ERROR_DEMO/demo(2)/demo/src/test/java/com/example/demo/ContractTest.java:[22,20] cannot find symbol
[ERROR]   symbol:   class ContractSub
[ERROR]   location: class com.example.demo.ContractTest
[ERROR] /D:/SPECMATIC_ERROR_DEMO/demo(2)/demo/src/test/java/com/example/demo/ContractTest.java:[27,21] cannot find symbol
[ERROR]   symbol:   method createStub(java.lang.String,int)
[ERROR]   location: class com.example.demo.ContractTest
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.

**Comment by harikrishnan83** [comment, marked as answer]
hello @prateekkarmakar , thanks for reaching out.

Specmatic 2.x onwards the minimum Java version required is 17. And that is why you are seeing this error, as you have rightly pointed out.

Since your project is in Java 11, the recommended approach here will be run Contract Test externally. By which I mean, instead of adding Specmatic as a library dependency in your project, you can run Specmatic contract test independently from using the [Specmatic JAR executable](https://specmatic.io/download.html), [NPM module](https://www.npmjs.com/package/specmatic) or [Docker image](https://hub.docker.com/r/znsio/specmatic). This way Specmatic is only interacting with your application over HTTP protocol and so Java versions will not matter.

You can follow [Java JAR, NPM, Docker, etc. related documentation](https://specmatic.io/getting_started.html) in our getting started page for the same. Hope this helps.

**Reply by prateekkarmakar** [reply]
@harikrishnan83 ,Thanks for the reply, will surely look into the documents. Is there any plan in future to support older versions of java (<17) where specmatic can run internally?

**Reply by harikrishnan83** [reply]
hello @prateekkarmakar, the minimum version we require for using Specmatic as a JAR library will be Java 17. There are several issues in support older versions of Java and simultaneously handling new versions of other framework libraries.

That said this should be an issue since you can always run Specmatic as a executable externally as I mentioned above. Thanks.
